src/zope/i18n/format.py

```diff
@@ -659,17 +666,18 @@ def buildDateTimeInfo(dt, calendar, pattern):
     # Getting the timezone right
     tzinfo = dt.tzinfo or pytz.utc
     tz_secs = tzinfo.utcoffset(dt).seconds
-    tz_secs = (tz_secs > 12*3600) and tz_secs-24*3600 or tz_secs
+    tz_secs = tz_secs - 24 * 3600 if tz_secs > 12 * 3600 else tz_secs
```

Comment:
(Unrelated to this PR) I have doubts about the correctness of this logic.  Time zones [UTC+12:45, UTC+13:00, UTC+14:00 exist](https://en.wikipedia.org/wiki/List_of_UTC_time_offsets#UTC+12:45,_M*).